### DISCUSSION TITLE
Is it possible to import faces from Synology Photos into Immich?

### DISCUSSION THREAD

**Discussion by hunorbatori** [immich-app/immich#3534] [Q&A] [answered]
Hi all,

Currently I'm using Synology Photos to manage my photo and video collection (around 25k pictures and videos combined) but considering to switch over to Immich.
A few months ago Synology introduced facial recognition to SPhotos which is good but far from perfect (none of it is I guess). It took some time to merge all the faces it considered as different person but was not, name them etc. This collection contains photos across multiple decades so many family members are children one some but grew into adults over time and ofc. the model had a hard time matching those but goofy facial expressions, sunglasses, face mask etc. also tricked it. Thus every people had multiple faces recognized, some of us (very close family members with a great number of photos) had more than 20 different face istances recognized.

It would be a real PITA to start over and name every face again and merge them together so is there any way I could migrate the faces from Synology Photos into Immich the way it would keep the already tagged faces and names and only scan for untagged faces on old photos and normal tagging on new photos?

Thanks!

**Comment by alextran1502** [comment, marked as answer]
Hello, there is not a way at the moment

**Reply by hunorbatori** [reply]
Thanks for the quick reply.
After further investigating the issue I found that Synology Photos store all kinds of metadata in the NASs internal PostgreSQL database which can be accessed.

Using this in an automatic way is not viable I know but If I can extract the data and transform it could be used somehown manually for a one time migration doesn't it?

I came up with two different ways of migrating tagging metadata from SPhotos to Immich. They are just quick toughts but I'm happy to research it more deeply if you think one of it can be viable.

So my ideas:
1. Exporting the tagging data from db to a plain file like CSV than writing a script to write the tagging data into the pictures files itself (If I recall correctly image files have fields to store tagging metadata). Than Immich can read those embended tagging metadata to populate its internal database.
2. Judging by the docker-compose of the project Immich uses PostgreSQL too to store data. If tagging metadata is stored in the db just like SPhotos does than in theory one could create an ETL process extract, transform and load the tagging data from SPhotos to Immich.

**Reply by alextran1502** [reply]
Hello, 

The face in Immich also has embedded vector data for the clustering mechanism. If you include data from Synology which I assume is just the coordinate of the faces, it won't be enough to fill the row of the table because the embedded vector data is required

**Reply by jrasm91** [reply]
If a file has a tag for people we could use that for a lookup instead of a nearest neighbor via a vector search. We should still compute the embedding either way.

This doesn't work when the image has multiple faces though unless there is some way to know which one belongs to which labeled person.

**Reply by hunorbatori** [reply]
> The face in Immich also has embedded vector data for the clustering mechanism.

I'm not familiar enough with Immich to know what this exactly means but I would need to do more research on both products to make an ETL anyway.

But consider how much time it would probably take to learn both SPhotos's and Immich's data structure and write an ETL that half a day of merging and naming faces again doesn't seem that much anymore 😅

Thank you for the fast and helpful support.

**Reply by joe128** [reply]
I'm also interested in a solution and i did a little research.
A query which displays the coordinates and the person-name can be as followed:
```SELECT p.name as person_name, fold.name as folder_name, fil.filename, fac.* 
FROM "public"."person" p,
  "public"."face" fac,
  "public"."unit" fil,
  "public"."folder" fold
WHERE fac.id_person = p.id
  AND fac.id_unit = fil.id
  AND fil.id_folder = fold.id
  AND p.name  = 'Person-Name'
  AND fold.name = '/folderUnderPhoto/...'
  AND fil.filename = '...';```
